### ISSUE TITLE
Question on reward_tracking_demo_key_body

### ISSUE THREAD

**Issue by cxr0726** [jimazeyu/exbody2#8]
Thanks for the excellent work and open-souce code!
I have some questions on the reward ```_reward_tracking_demo_key_body```. 

    def _reward_tracking_demo_key_body(self):
        demo_key_body_pos_local = self._curr_demo_keybody.view(self.num_envs, self._num_key_bodies, 3)
        if self.cfg.motion.global_keybody:
            curr_demo_xyz = torch.cat((self.target_pos_abs, self._curr_demo_root_pos[:, 2:3]), dim=-1)
        else:
            curr_demo_xyz = torch.cat((self.root_states[:, :2], self._curr_demo_root_pos[:, 2:3]), dim=-1)
        demo_global_body_pos = local_to_global(self._curr_demo_quat, demo_key_body_pos_local, curr_demo_xyz).view(self.num_envs, -1)
        cur_global_body_pos = self.rigid_body_states[:, self._key_body_ids_sim[self._key_body_ids_sim_subset], :3].view(self.num_envs, -1)

        rew = torch.exp(-torch.norm(cur_global_body_pos - demo_global_body_pos, dim=1))
        return rew

If I understand correctly, ```demo_key_body_pos_local``` represents the pose of each key body in the reference motion coordinate system, and it can be expressed as 

> $p^{demo\ local}=q^{-1}(p^{ref}-r^{ref})$

wher $q,p^{ref},r^{ref}$ correspond to ```global_ref_root_quat```, ```global_ref_keybody_pos```, ```global_ref_root_xyz```, respectively.

Then if well-trained, ```self.root_states[:, 3]``` should be very close to ```self._curr_demo_root_pos[:, 2:3]```. Therefore, ```curr_demo_xyz``` can be written as ```self.root_states[:, 3]```, denoted as $r^{real}$.

So ```demo_global_body_pos ``` can be expressed as 

> $p^{demo\ global}=q(p^{demo\ local}+r^{real})$

By integrating the above equations, we can obtain

> $p^{demo\ global}=p^{ref}-r^{ref}+r^{real}$

The error term ```cur_global_body_pos - demo_global_body_pos```   can thus be written as $(p^{ref}-r^{ref})-(p^{real}-r^{real})$, where $p^{real}$ corresponds to  ```global_real_keybody_pos```. Driving this error toward zero will cause the robot's forward orientation to approach that of the reference motion in the world coordinate system, which seems somewhat conflicting with tracking keypoints in the local frame.

The video below better illustrates my issue. When resetting the robot, I changed its orientation by modifying the initial quaternion from (0, 0, 0, 1) to (0, 0, 1, 0). When I run python play_priv.py using the provided example policy, the robot first turns around to align its forward direction with that of the reference motion before it starts to track the keypoints normally. However, the robot is supposed to track the keypoints in the local frame correctly from the beginning, regardless of its initial orientation.

So I would like to ask, when performing teleoperation in the real world, how are the aforementioned issues addressed? Are they resolved by using a large amount of diverse and wide-coverage reference data during training?

Thanks!

https://github.com/user-attachments/assets/46041192-6d5c-4d91-8cf7-3a9b1a002730

**Comment by jimazeyu** [comment]
You can regard it just a way of data augmentation to increase sampling efficiency. It's truly misleading of saying local tracking, it just works for training process for better learning results. For real deployment, we don't have any global reference. The trick will help robot understand the relationship between the joint positions and the keybody positions. If you try to train your robot with global keypoints, it means your robot need to estimate it's global state with just history info, which is really hard for now.

**Comment by jimazeyu** [comment]
BTW, when you are tracking a single motion(or some really small groups)  with global, use global tracking reward may lead to very great tracking result, it's possible that we train every motion seperately and distill them into one policy. But for now the normal training framework can't work well like training a avatar like in PHC, the reason maybe the capability of the robots.

**Comment by cxr0726** [comment]
I got it. Thank you for your patient reply. So when deploying in the real world, the robot's initial yaw angle will be aligned with the axis of the first frame (either the reference motion or the human operator). Is that correct?
